### PR TITLE
local docker environment

### PR DESCRIPTION
**Pull request by acidtib**
this is the first PR in the journey of mrsk

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -35,15 +35,15 @@ COPY . .
 RUN bundle exec bootsnap precompile app/ lib/
 
 # Precompiling assets for production without requiring secret RAILS_MASTER_KEY
-RUN SECRET_KEY_BASE=1 ./bin/rails assets:precompile
```
Comment (obie) [inline Dockerfile:38]: this has been working fine, sure about changing it?

File: config/boot.rb [hunk 2]
```diff
@@ -3,9 +3,3 @@
 require "bundler/setup" # Set up gems listed in the Gemfile.
 require "bootsnap/setup" # Speed up boot time by caching expensive operations.
```
Comment (obie) [inline config/boot.rb:5]: where does dotenv get loaded now?

File: docker-compose.yml [hunk 3]
```diff
@@ -44,9 +35,9 @@ services:
       interval: 5s
       timeout: 5s
       retries: 5
-    command: redis-server
+    command: redis-server --requirepass password
     ports:
-      - '6379:6379'
+      - '6380:6379'
```
Comment (obie) [inline docker-compose.yml:40]: what's this change about?

File: config/boot.rb (hunk 2 above)
Comment (acidtib) [reply config/boot.rb:5]: the environment is loaded using docker-compose the .env.local

File: docker-compose.yml (hunk 3 above)
Comment (acidtib) [reply docker-compose.yml:40]: this is to prevent conflicts on a local computer that may already have PG or Redis running

File: Dockerfile (hunk 1 above)
Comment (acidtib) [reply Dockerfile:38]: SECRET_KEY_BASE_DUMMY works now since we are loading rails 7.1 options

**Comment by obie** [comment]
Thanks for the PR and all the help this evening.

**Pull request merged**